File: archive/legacy-setup-2026-05-13/generate_inkognito_icons.swift
import AppKit

struct OpticalProfile {
    let topBarWidth: CGFloat
    let topBarHeight: CGFloat
    let stemWidth: CGFloat
    let stemHeight: CGFloat
    let topBarCornerRadius: CGFloat
    let stemCornerRadius: CGFloat

    static func forCanvas(_ size: Int) -> OpticalProfile {
        switch size {
        case ..<32:
            return OpticalProfile(
                topBarWidth: 80,
                topBarHeight: 22,
                stemWidth: 36,
                stemHeight: 62,
                topBarCornerRadius: 0,
                stemCornerRadius: 0
            )
        case ..<64:
            return OpticalProfile(
                topBarWidth: 72,
                topBarHeight: 20,
                stemWidth: 32,
                stemHeight: 62,
                topBarCornerRadius: 0,
                stemCornerRadius: 0
            )
        case ..<128:
            return OpticalProfile(
                topBarWidth: 68,
                topBarHeight: 18,
                stemWidth: 28,
                stemHeight: 62,
                topBarCornerRadius: 1,
                stemCornerRadius: 1
            )
        case ..<256:
            return OpticalProfile(
                topBarWidth: 64,
                topBarHeight: 16,
                stemWidth: 24,
                stemHeight: 60,
                topBarCornerRadius: 1,
                stemCornerRadius: 1
            )
        default:
            return OpticalProfile(
                topBarWidth: 64,
                topBarHeight: 14,
                stemWidth: 20,
                stemHeight: 60,
                topBarCornerRadius: 2,
                stemCornerRadius: 3
            )
        }
    }
}

enum Theme {
    case light
    case dark

    var startColor: NSColor {
        switch self {
        case .light: return NSColor(calibratedRed: 1.0, green: 1.0, blue: 1.0, alpha: 1.0)
        case .dark: return NSColor(calibratedRed: 0.1725, green: 0.1725, blue: 0.1804, alpha: 1.0)
        }
    }

    var endColor: NSColor {
        switch self {
        case .light: return NSColor(calibratedRed: 0.898, green: 0.898, blue: 0.9176, alpha: 1.0)
        case .dark: return NSColor(calibratedRed: 0.1098, green: 0.1098, blue: 0.1176, alpha: 1.0)
        }
    }

    var glyphColor: NSColor {
        switch self {
        case .light: return NSColor(calibratedRed: 0.0392, green: 0.0392, blue: 0.0471, alpha: 1.0)
        case .dark: return NSColor(calibratedRed: 0.9608, green: 0.9608, blue: 0.9686, alpha: 1.0)
        }
    }
}

@discardableResult
func renderIcon(size: Int, theme: Theme, outputURL: URL) throws -> Bool {
    let image = NSImage(size: NSSize(width: size, height: size))

    image.lockFocus()
    defer { image.unlockFocus() }

    NSGraphicsContext.current?.imageInterpolation = .high

    let canvas = NSRect(x: 0, y: 0, width: CGFloat(size), height: CGFloat(size))
    let squircleInset = CGFloat(size) * 0.033
    let iconRect = canvas.insetBy(dx: squircleInset, dy: squircleInset)
    let radius = iconRect.width * 0.185
    let squirclePath = NSBezierPath(roundedRect: iconRect, xRadius: radius, yRadius: radius)

    squirclePath.addClip()

    let gradient = NSGradient(starting: theme.startColor, ending: theme.endColor)!
    gradient.draw(in: squirclePath, angle: -90)

    let shadow = NSShadow()
    shadow.shadowColor = NSColor.black.withAlphaComponent(theme == .light ? 0.12 : 0.36)
    shadow.shadowBlurRadius = CGFloat(size) * 0.045
    shadow.shadowOffset = NSSize(width: 0, height: -CGFloat(size) * 0.01)
    shadow.set()

    let borderPath = NSBezierPath(roundedRect: iconRect, xRadius: radius, yRadius: radius)
    theme.glyphColor.withAlphaComponent(theme == .light ? 0.05 : 0.10).setStroke()
    borderPath.lineWidth = max(1, CGFloat(size) * 0.004)
    borderPath.stroke()

    let profile = OpticalProfile.forCanvas(size)
    let scale = iconRect.width / 140.0

    let originX = iconRect.minX
    let originY = iconRect.minY

    func scaledRect(x: CGFloat, y: CGFloat, width: CGFloat, height: CGFloat) -> NSRect {
        NSRect(
            x: originX + x * scale,
            y: originY + y * scale,
            width: width * scale,
            height: height * scale
        )
    }

    let topBarX = 70.0 - (profile.topBarWidth / 2.0)
    let topBarRect = scaledRect(
        x: topBarX,
        y: 32,
        width: profile.topBarWidth,
        height: profile.topBarHeight
    )
    let stemX = 70.0 - (profile.stemWidth / 2.0)
    let stemRect = scaledRect(
        x: stemX,
        y: 58,
        width: profile.stemWidth,
        height: profile.stemHeight
    )

    theme.glyphColor.setFill()
    NSBezierPath(
        roundedRect: topBarRect,
        xRadius: profile.topBarCornerRadius * scale,
        yRadius: profile.topBarCornerRadius * scale
    ).fill()
    NSBezierPath(
        roundedRect: stemRect,
        xRadius: profile.stemCornerRadius * scale,
        yRadius: profile.stemCornerRadius * scale
    ).fill()

    guard
        let tiffData = image.tiffRepresentation,
        let rep = NSBitmapImageRep(data: tiffData),
        let pngData = rep.representation(using: .png, properties: [:])
    else {
        return false
    }

    try pngData.write(to: outputURL, options: .atomic)
    return true
}

let fileManager = FileManager.default
let root = URL(fileURLWithPath: fileManager.currentDirectoryPath)
let appIconDir = root.appendingPathComponent("HideMyData/Assets.xcassets/AppIcon.appiconset", isDirectory: true)
let appLogoDir = root.appendingPathComponent("HideMyData/Assets.xcassets/AppLogo.imageset", isDirectory: true)

let outputs: [(String, Int, Theme)] = [
    ("icon_16.png", 16, .light),
    ("icon_16@2x.png", 32, .light),
    ("icon_32.png", 32, .light),
    ("icon_32@2x.png", 64, .light),
    ("icon_128.png", 128, .light),
    ("icon_128@2x.png", 256, .light),
    ("icon_256.png", 256, .light),
    ("icon_256@2x.png", 512, .light),
    ("icon_512.png", 512, .light),
    ("icon_512@2x.png", 1024, .light),
    ("icon_1024.png", 1024, .light),
    ("icon_1024_dark.png", 1024, .dark),
    ("logo.png", 1024, .light)
]

for (filename, size, theme) in outputs {
    let directory = filename == "logo.png" ? appLogoDir : appIconDir
    let outputURL = directory.appendingPathComponent(filename)
    try renderIcon(size: size, theme: theme, outputURL: outputURL)
    print("Wrote \(outputURL.path)")
}

File: archive/legacy-setup-2026-05-13/generate_inkognito_icons.py
from pathlib import Path

from PIL import Image, ImageDraw


ROOT = Path(__file__).resolve().parent.parent
APPICON_DIR = ROOT / "HideMyData" / "Assets.xcassets" / "AppIcon.appiconset"
APPLOGO_DIR = ROOT / "HideMyData" / "Assets.xcassets" / "AppLogo.imageset"


def optical_profile(size: int) -> dict[str, float]:
    if size < 32:
        return dict(top_w=84, top_h=26, stem_w=40, stem_h=66, top_r=2, stem_r=4, top_x=28, top_y=25, stem_x=50, stem_y=54)
    if size < 64:
        return dict(top_w=74, top_h=22, stem_w=32, stem_h=64, top_r=2, stem_r=3, top_x=33, top_y=27, stem_x=54, stem_y=55)
    if size < 128:
        return dict(top_w=70, top_h=18, stem_w=28, stem_h=62, top_r=2, stem_r=3, top_x=35, top_y=30, stem_x=56, stem_y=56)
    if size < 256:
        return dict(top_w=68, top_h=16, stem_w=24, stem_h=60, top_r=2, stem_r=3, top_x=36, top_y=31, stem_x=58, stem_y=58)
    return dict(top_w=64, top_h=15 if size == 256 else 14, stem_w=20, stem_h=60, top_r=2, stem_r=3, top_x=38, top_y=32, stem_x=60, stem_y=58)


def interpolate(c1, c2, t: float):
    return tuple(round(a + (b - a) * t) for a, b in zip(c1, c2))


def render_icon(size: int, dark: bool = False) -> Image.Image:
    base = Image.new("RGBA", (size, size), (0, 0, 0, 0))

    rect = (
        round(size * 5 / 140),
        round(size * 5 / 140),
        round(size * 135 / 140),
        round(size * 135 / 140),
    )
    radius = round(size * 25 / 140)

    start = (44, 44, 46) if dark else (255, 255, 255)
    end = (28, 28, 30) if dark else (229, 229, 234)
    glyph = (245, 245, 247, 255) if dark else (10, 10, 12, 255)
    mask = Image.new("L", (size, size), 0)
    ImageDraw.Draw(mask).rounded_rectangle(rect, radius=radius, fill=255)

    gradient = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    grad_px = gradient.load()
    for y in range(size):
        t = y / max(size - 1, 1)
        color = interpolate(start, end, t)
        for x in range(size):
            grad_px[x, y] = (*color, 255)
    gradient.putalpha(mask)
    base.alpha_composite(gradient)

    draw = ImageDraw.Draw(base)
    prof = optical_profile(size)
    scale = (rect[2] - rect[0]) / 140.0
    ox, oy = rect[0], rect[1]

    def scaled_box(x, y, w, h):
        return (
            round(ox + x * scale),
            round(oy + y * scale),
            round(ox + (x + w) * scale),
            round(oy + (y + h) * scale),
        )

    top_box = scaled_box(prof["top_x"], prof["top_y"], prof["top_w"], prof["top_h"])
    stem_box = scaled_box(prof["stem_x"], prof["stem_y"], prof["stem_w"], prof["stem_h"])

    draw.rounded_rectangle(top_box, radius=prof["top_r"] * scale, fill=glyph)
    draw.rounded_rectangle(stem_box, radius=prof["stem_r"] * scale, fill=glyph)
    return base


OUTPUTS = [
    ("icon_16.png", 16, False),
    ("icon_16@2x.png", 32, False),
    ("icon_32.png", 32, False),
    ("icon_32@2x.png", 64, False),
    ("icon_128.png", 128, False),
    ("icon_128@2x.png", 256, False),
    ("icon_256.png", 256, False),
    ("icon_256@2x.png", 512, False),
    ("icon_512.png", 512, False),
    ("icon_512@2x.png", 1024, False),
    ("icon_1024.png", 1024, False),
    ("icon_1024_dark.png", 1024, True),
]


for filename, size, dark in OUTPUTS:
    render_icon(size, dark).save(APPICON_DIR / filename)
    print(f"Wrote {(APPICON_DIR / filename).relative_to(ROOT)}")

render_icon(1024, False).save(APPLOGO_DIR / "logo.png")
print(f"Wrote {(APPLOGO_DIR / 'logo.png').relative_to(ROOT)}")
